synapse/storage/databases/main/events_worker.py

```diff
@@ -1791,10 +1791,10 @@ def get_ex_outlier_stream_rows_txn(
                 " LEFT JOIN event_relations USING (event_id)"
                 " LEFT JOIN room_memberships USING (event_id)"
                 " LEFT JOIN rejections USING (event_id)"
-                " WHERE ? < event_stream_ordering"
-                " AND event_stream_ordering <= ?"
+                " WHERE ? < out.event_stream_ordering"
+                " AND out.event_stream_ordering <= ?"
                 " AND out.instance_name = ?"
-                " ORDER BY event_stream_ordering ASC"
+                " ORDER BY out.event_stream_ordering ASC"
```

Comment:
This disambiguates the query. Unfortunately it means that old old Synapses (without this code change) will fail to run this query against an upgraded database.

Not sure how we should handle this.

Reply:
https://github.com/matrix-org/synapse/issues/15014